react-components/src/query/useSearchMappedEquipmentAssetMappings.tsx

```diff
@@ -100,21 +138,46 @@ export const useAllMappedEquipmentAssetMappings = (
 
         const mappings = await sdk.assetMappings3D.filter(model.modelId, model.revisionId, {
           cursor: nextCursor === 'start' ? undefined : nextCursor,
-          limit: 1000
+          limit
         });
 
+        usedCursors.current.add(nextCursor);
+
         return { mappings, model };
       });
 
       const currentPagesOfAssetMappings = await Promise.all(currentPagesOfAssetMappingsPromises);
 
       const modelsAssets = await getAssetsFromAssetMappings(sdk, currentPagesOfAssetMappings);
+      const nextCursors = currentPagesOfAssetMappings
+        .map(({ mappings }) => mappings.nextCursor)
+        .filter(isDefined);
 
-      return modelsAssets;
+      return await Promise.resolve({
+        modelsAssets,
+        nextCursors
+      });
     },
     initialPageParam: models.map((model) => ({ cursor: 'start', model })),
     staleTime: Infinity,
-    getNextPageParam
+    getNextPageParam: (lastPage: {
+      modelsAssets: ModelMappingsWithAssets[];
+      nextCursors: string[];
+    }): Array<{ cursor: string | undefined; model: AddModelOptions }> | undefined => {
+      const nextCursors = lastPage.nextCursors
```

Comment:
I would prefer to be more explicit in var names. But this is more a personal taste :) 

For example: as this const is holding not only the cursor but also the model props, maybe the name could be something like: `nextCursorAndModelOptions`
Also, the returning I think could be a type as well:
`NextCursorAndModelOptions[]  | undefined`

Reply:
return `type` should match the `query` return type, so there is not much we can do to change it.

Reply:
Hmmm yeah. I see